src/analytics/addPageAction.ts

```diff
@@ -0,0 +1,49 @@
+import NewRelicAPI from "new-relic-browser";
+import { parseQueryString } from "utils";
+
+declare global {
+  interface Window {
+    newrelic: typeof NewRelicAPI; // eslint-disable-line no-undef
+  }
+}
+
+export interface Analytics<Action> {
+  sendEvent: (action: Action) => void;
+}
+
+type AnalyticsObject =
+  | "Patch"
+  | "Task"
+  | "Navbar"
+  | "Breadcrumb"
+  | "UserPatches"
+  | "CommitQueue"
+  | "Configure";
+
+interface RequiredProperties {
+  object: AnalyticsObject;
+  userId: string;
+}
+interface ActionType {
+  name: string;
+}
+
+export interface Properties {
+  [key: string]: string | number;
+}
+
+export const addPageAction = <A extends ActionType, P extends Properties>(
+  { name, ...actionProps }: A,
+  properties: P & RequiredProperties
+) => {
+  const { newrelic } = window;
+  if (typeof newrelic !== "object") {
+    return;
+  }
+  const { search } = window.location;
```

Comment:
can we use useLocation() in react router for this? You might need to pass it from the usePatchAnalytics() hook

Reply:
we cannot because this is a function, not a hook, and therefore cannot use hooks.

Reply:
also, the useLocation in the hook at the time this function called is the previous value. we need the value of search at the time the function is called to be the most up to date.